shared/app/controllers/concerns/localization.rb

```diff
@@ -2,12 +2,18 @@ module Localization
   extend ActiveSupport::Concern
 
   included do
-    before_action :set_locale
+    around_action :switch_locale
 
     protected
 
-    def set_locale
-      I18n.locale = params[:locale] || I18n.default_locale
+    def switch_locale(&action)
+      locale = extract_locale_from_param || I18n.default_locale
+
+      I18n.with_locale(locale, &action)
+    end
+
+    def extract_locale_from_param
+      I18n.available_locales.include?(params[:locale]&.to_sym) ? params[:locale].to_sym : nil
```

Comment:
Not a big thing, but can we use `I18n.locale_available?` instead.

Reply:
Good to know that @Tolsee 👍 Will change it

Reply:
Changed that @Tolsee 👍